flink-connectors/flink-connector-kafka/src/main/java/org/apache/flink/connector/kafka/sink/KafkaSerializerWrapper.java

```diff
@@ -65,8 +77,15 @@ public void open(InitializationContext context) throws Exception {
                             serializerClass.getName(),
                             Serializer.class,
                             getClass().getClassLoader());
+
+            if (config.isEmpty()) {
```

Comment:
Is it possible that user's deserializer::configure(...) could still be useful (e.g. has side effects) even when the config is empty?

Would it be simpler and safer to remove this check?

Reply:
I have thought about it. The question is why should user call `configure(...)` with an empty config?

Allowing empty config going through sounds like a backdoor that the Map reference is injected into `configure(...)`. User might want to change the content of the map outside of the serializer/deserializer and have effect on exist serializer/deserializer retroactively(hacking, which is not a good idea), if the implementation of serializer/deserializer is not done correctly. The config map as input should be used as a one-off source and should be even turned into an immutable map before hand it over to the `Configurable/Deserializer::configure(...)` call.

User will be aware that nothing will be done with an empty config where e.g. content were forgot to add by mistake. How about adding logger so that any potential issues will be trackable in the log files.

Reply:
It is possible that relying on an invocation of `configure(...)` with an empty config is a bad practice. However, since this is an existing API that public users might already be using, users will be unhappy if our change break their existing application, right? I am just not sure it is a good idea to tell users that we break their application because we believe their usage is wrong. 

Also, the public method will probably be easier to understand/use if its semantics is `deserializer::configure(...) will be invoked with the config map provided`, rather than saying `deserializer::configure(...) will be invoked with the config map provided if the map is empty`. What do you think?

Reply:
fair enough, I will remove this part.